lib/config/presets/index.ts

```diff
@@ -378,11 +394,81 @@ export async function resolveConfigPresets(
         val as RenovateConfig,
         baseConfig,
         ignorePresets,
-        existingPresets
+        existingPresets,
+        shallowLog === 'off' ? 'off' : 'resolveOnly',
+        shallowExtendedConfig
       );
     }
   }
   logger.trace({ config: inputConfig }, 'Input config');
   logger.trace({ config }, 'Resolved config');
   return config;
 }
+
+export function updateShallowConfig(
+  shallowExtendedConfig: Record<string, RenovateConfig>,
+  fetchedPreset: RenovateConfig
+): void {
+  // save extends array from both config
+  const tempExtend = new Set<string>();
+  if (is.array(shallowExtendedConfig.value?.extends)) {
+    for (const extend of shallowExtendedConfig.value.extends) {
+      tempExtend.add(extend);
+    }
+  }
+  if (is.array(fetchedPreset?.extends)) {
+    for (const extend of fetchedPreset.extends) {
+      tempExtend.add(extend);
+    }
+  }
+  // merge config
+  shallowExtendedConfig.value = mergeChildConfig(
+    shallowExtendedConfig.value,
+    fetchedPreset
+  );
+  // add extends back to the result
+  if (tempExtend.size && shallowExtendedConfig.value.extends) {
+    shallowExtendedConfig.value.extends = Array.from(tempExtend);
+  }
+}
+
+export function isPresetForShallowConfig(presetSource: string): boolean {
+  if (whitesourcePresetRegex.test(presetSource)) {
+    return false;
+  }
+  return (
+    presetSource.startsWith('github>') ||
+    presetSource.startsWith('gitlab>') ||
+    presetSource.startsWith('gitea>') ||
+    presetSource.startsWith('local>')
+  );
+}
```

Comment:
```suggestion
export function isPresetForShallowConfig(presetSource: string): boolean {
  return whitesourcePresetRegex.test(presetSource);
}
```

Using the suggested regex, this function can be simpler.

Reply:
That regex is meant for presets that we wish to not include for logging. 
Now we are skipping config under whitesource such as `github>whitesource/merge-confidence:beta`

Reply:
oh, right. my suggestion would alter the logic. missed the opposite return value.